test/integration/puppetlabs/trapperkeeper/services/scheduler/scheduler_service_test.clj

```diff
@@ -62,37 +73,185 @@
+            (stop-job service job-1)
+            (Thread/sleep 100)
+            (is (= 1 (count-jobs service)))
+            (stop-job service job-2)
+            (Thread/sleep 100)
+            (is (= 0 (count-jobs service)))))))))
+
+  (testing "count-jobs shows correct number of group-id and non-group-id jobs"
+    (with-app-with-empty-config app [scheduler-service]
+      (let [service (tk/get-service app :SchedulerService)
+            interval 10
+            job-0 (guaranteed-start-interval-job service interval)
+            group-id :unique-group-id]
+        (is (= 1 (count-jobs service)))
+        (let [group-id-job-0 (guaranteed-start-interval-job service interval group-id)]
+          (is (= 2 (count-jobs service)))
+          (is (= 1 (count-jobs service group-id)))
+          (let [job-1 (guaranteed-start-interval-job service interval)]
+            (is (= 3 (count-jobs service)))
+            (is (= 1 (count-jobs service group-id)))
+              (let [group-id-job-1 (guaranteed-start-interval-job service interval group-id)]
+                (is (= 4 (count-jobs service)))
+                (is (= 2 (count-jobs service group-id)))
+                (let [job-2 (guaranteed-start-interval-job service interval)]
+                  (is (= 5 (count-jobs service)))
+                  (is (= 2 (count-jobs service group-id)))
+                  ; stop the first job and wait a bit
+                  (stop-job service job-0)
+                  (Thread/sleep 100)
+                  (is (= 4 (count-jobs service)))
+                  (is (= 2 (count-jobs service group-id)))
+                  (stop-job service group-id-job-0)
+                  (Thread/sleep 100)
+                  (is (= 3 (count-jobs service)))
+                  (is (= 1 (count-jobs service group-id)))
+                  (stop-job service job-1)
+                  (Thread/sleep 100)
+                  (is (= 2 (count-jobs service)))
+                  (is (= 1 (count-jobs service group-id)))
+                  (stop-job service group-id-job-1)
+                  (Thread/sleep 100)
+                  (is (= 1 (count-jobs service)))
+                  (is (= 0 (count-jobs service group-id)))
+                  (stop-job service job-2)
+                  (Thread/sleep 100)
+                  (is (= 0 (count-jobs service)))
+                  (is (= 0 (count-jobs service group-id))))))))))
+
+  (testing "after reduces count when complete"
+    (with-app-with-empty-config app [scheduler-service]
+      (let [service (tk/get-service app :SchedulerService)
+            delay 100
+            wait-for-start (promise)
+            completed (promise)
+            job (fn []
+                  (deref wait-for-start)
+                  (deliver completed (System/currentTimeMillis)))]
+        (after service delay job)
+        (is (= 1 (count-jobs service)))
+        (deliver wait-for-start true)
+        (deref completed)
+        (Thread/sleep 100)
```

Comment:
The job could have completed, but not removed from the set. There is a small window (that finally block in the "after" function) that needs to get executed before it is really removed, so I think it is needed.